app/ssttpcalculator/CalculatorService.scala

```diff
@@ -417,9 +418,9 @@ object CalculatorService {
       if (initialPayment.equals(BigDecimal(0))) {
         val daysBetweenStartAndPaymentDates = DAYS.between(startDate, defaultPaymentDate)
 
-        if (daysBetweenStartAndPaymentDates < oneWeek && DAYS.between(startDate, defaultPaymentDatePlusOneMonth) < oneWeek)
+        if (daysBetweenStartAndPaymentDates < tenDays && DAYS.between(startDate, defaultPaymentDatePlusOneMonth) < oneWeek)
           (defaultPaymentDatePlusTwoMonths, defaultEndDatePlusTwoMonths)
-        else if (daysBetweenStartAndPaymentDates < oneWeek)
+        else if (daysBetweenStartAndPaymentDates < tenDays)
           (defaultPaymentDatePlusOneMonth, defaultEndDatePlusOneMonth)
         else
           (defaultPaymentDate, defaultEndDate)
```

Comment:
Nb: Not sure if it's relevant, but there's references still to 7 days (a week) for the case where there is an upfront payment, down on lines 428+

Reply:
Yes, that is true. But for such cases it worked fine because this start date was altered later in the process. 
I'll have a look anyway.